src/app/clusters/time-synchronization-server/time-synchronization-server.cpp

```diff
@@ -0,0 +1,685 @@
+{
+    CHIP_ERROR err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        DataModel::List<TimeSynchronization::Structs::TimeZoneStruct::Type> tzList =
+            TimeSynchronizationServer::Instance().GetTimeZone();
+        for (auto it = tzList.begin(); it != tzList.end(); ++it)
+        {
+            ReturnErrorOnFailure(encoder.Encode(TimeZoneCodec(*it)));
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+    return err;
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::ReadDSTOffset(EndpointId endpoint, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = aEncoder.EncodeList([](const auto & encoder) -> CHIP_ERROR {
+        DataModel::List<TimeSynchronization::Structs::DSTOffsetStruct::Type> dst =
+            TimeSynchronizationServer::Instance().GetDSTOffset();
+        for (auto it = dst.begin(); it != dst.end(); ++it)
+        {
+            ReturnErrorOnFailure(encoder.Encode(*it));
+        }
+
+        return CHIP_NO_ERROR;
+    });
+
+    return err;
+}
+
+CHIP_ERROR TimeSynchronizationAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    if (aPath.mClusterId != TimeSynchronization::Id)
+    {
+        return CHIP_ERROR_INVALID_PATH_LIST;
+    }
+
+    switch (aPath.mAttributeId)
+    {
+    case UTCTime::Id: {
+        DataModel::Nullable<uint64_t> currentUtcTime;
+        UTCTime::Get(aPath.mEndpointId, currentUtcTime);
+        if (!currentUtcTime.IsNull())
+        {
+            System::Clock::Microseconds64 utcTime;
+            System::SystemClock().GetClock_RealTime(utcTime);
+            UTCTime::Set(aPath.mEndpointId, utcTime.count());
+            return aEncoder.Encode(utcTime.count());
+        }
+        else
+        {
+            return aEncoder.EncodeNull();
+        }
+    }
+    case TrustedTimeSource::Id: {
+        return ReadTrustedTimeSource(aPath.mEndpointId, aEncoder);
+    }
+    case DefaultNTP::Id: {
+        return ReadDefaultNtp(aPath.mEndpointId, aEncoder);
+    }
+    case TimeZone::Id: {
+        return ReadTimeZone(aPath.mEndpointId, aEncoder);
+    }
+    case DSTOffset::Id: {
+        return ReadDSTOffset(aPath.mEndpointId, aEncoder);
+    }
+    case TimeZoneListMaxSize::Id: { // can't find a way to initialize default value for F quality
+        uint8_t max = CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE;
+        TimeZoneListMaxSize::Set(aPath.mEndpointId, CHIP_CONFIG_TIME_ZONE_LIST_MAX_SIZE);
+        return aEncoder.Encode(max);
+    }
+    case DSTOffsetListMaxSize::Id: {
+        uint8_t max = CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE;
+        DSTOffsetListMaxSize::Set(aPath.mEndpointId, CHIP_CONFIG_DST_OFFSET_LIST_MAX_SIZE);
+        return aEncoder.Encode(max);
+    }
+    case LocalTime::Id: {
+        DataModel::Nullable<uint64_t> localTime;
+        if (computeLocalTime(aPath.mEndpointId))
+        {
+            LocalTime::Get(aPath.mEndpointId, localTime);
```

Comment:
Why are we storing any of this in the attribute store?

Reply:
please elaborate

Reply:
I think the real issue here is that we have a really complex system for handling attributes and we don't have nearly enough documentation on how to actually choose between these. We should really try to put something more formal together to explain how everything fits together and falls through and hangs with the zap storage options, but as with everything, finding the time required to do that is  challenging.